tests/components/hvv_departures/test_config_flow.py

```diff
@@ -0,0 +1,380 @@
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+        assert result_user["type"] == "form"
+        assert result_user["errors"] == {"base": "cannot_connect"}
+
+
+async def test_user_flow_station(hass):
+    """Test that config flow handles empty data on step station."""
+
+    with patch(
+        "homeassistant.components.hvv_departures.hub.GTI.init", return_value=True,
+    ), patch(
+        "pygti.gti.GTI.checkName", return_value={"returnCode": "OK", "results": []},
+    ):
+
+        # step: user
+
+        result_user = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data={
+                CONF_HOST: "api-test.geofox.de",
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+        assert result_user["step_id"] == "station"
+
+        # step: station
+        result_station = await hass.config_entries.flow.async_configure(
+            result_user["flow_id"], None,
+        )
+        assert result_station["type"] == "form"
+        assert result_station["step_id"] == "station"
+
+
+async def test_user_flow_station_select(hass):
+    """Test that config flow handles empty data on step station_select."""
+
+    with patch(
+        "homeassistant.components.hvv_departures.hub.GTI.init", return_value=True,
+    ), patch(
+        "pygti.gti.GTI.checkName", return_value=FIXTURE_CHECK_NAME,
+    ):
+        result_user = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data={
+                CONF_HOST: "api-test.geofox.de",
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+        result_station = await hass.config_entries.flow.async_configure(
+            result_user["flow_id"], {CONF_STATION: "Wartenau"},
+        )
+
+        # step: station_select
+        result_station_select = await hass.config_entries.flow.async_configure(
+            result_station["flow_id"], None,
+        )
+
+        assert result_station_select["type"] == "form"
+        assert result_station_select["step_id"] == "station_select"
+
+
+async def test_options_flow(hass):
+    """Test that options flow works."""
+
+    config_entry = MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="Wartenau",
+        data=FIXTURE_CONFIG_ENTRY,
+        source="user",
+        connection_class=CONN_CLASS_CLOUD_POLL,
+        system_options={"disable_new_entities": False},
+        options=FIXTURE_OPTIONS,
+        unique_id="1234",
+    )
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+
+    with patch(
+        "homeassistant.components.hvv_departures.async_setup_entry", return_value=True
```

Comment:
This patch has already been called at this point by `hass.config_entries.async_setup`.